Dockerfile

```diff
@@ -20,8 +20,24 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update -yqq && \
     python3-pylibacl \
     python3-pyxattr
 
+# Build dependencies specific for rdiff-backup development and testing
+RUN DEBIAN_FRONTEND=noninteractive apt-get update -yqq && \
+    apt-get install -y --no-install-recommends \
+    tox \
+    rdiff \
+    libacl1-dev # /usr/include/sys/acl.h
+ 
 # Build dev image
 # docker build --pull --tag rdiff-backup-dev:debian-sid .
 
-# Build rdiff-backup
-# docker run -it -v ${PWD}:/build -w /build rdiff-backup-dev:debian-sid ./setup.py build
+# Build rdiff-backup (assumes source is in directory 'rdiff-backup' and it's parent is writeable)
+# docker run -it -v ${PWD}/..:/build -w /build/rdiff-backup rdiff-backup-dev:debian-sid ./setup.py build
+
+# Run tests
+# docker run -it -v ${PWD}/..:/build -w /build/rdiff-backup rdiff-backup-dev:debian-sid bash
+#   cd ..
+#   curl -LO https://github.com/ericzolf/rdiff-backup/releases/download/Testfiles2019-08-10/rdiff-backup_testfiles_2019-08-10.tar.gz
+#   tar xvf *.tar.gz
```

Comment:
this step must be run as root else the files have the wrong permission

Reply:
Additionally you must run `sudo ./rdiff-backup_testfiles.fix.sh` to make sure the files not owned by root are owned by the current normal user, or you'll get strange errors. Pay attention, this step assumes that the normal user is documented in sudo's environment variables.

Reply:
The script outputs:
```
root@6471d534de6a:/build# ./rdiff-backup_testfiles.fix.sh
chown: invalid user: 'rdiff-backup_testfiles'
chown: invalid user: 'rdiff-backup_testfiles/restoretest3/rdiff-backup-data/increments/increment1/config.log.1970-01-01T09:20:00+01:00.snapshot.gz'
chown: invalid user: 'rdiff-backup_testfiles/bigdir/subdir1/subdir16/file4'
chown: invalid user: 'rdiff-backup_testfiles/bigdir/subdir0/subdir48/file47'
chown: invalid user: 'rdiff-backup_testfiles/bigdir/subdir2/subdir23/file3'
chown: invalid user: 'rdiff-backup_testfiles/bigdir/subdir3/subdir10/file31'
chgrp: invalid group: 'rdiff-backup_testfiles'
chgrp: invalid group: 'rdiff-backup_testfiles/restoretest3/rdiff-backup-data/increments/increment1/config.log.1970-01-01T09:20:00+01:00.snapshot.gz'
chgrp: invalid group: 'rdiff-backup_testfiles/bigdir/subdir1/subdir16/file4'
chgrp: invalid group: 'rdiff-backup_testfiles/bigdir/subdir0/subdir48/file47'
chgrp: invalid group: 'rdiff-backup_testfiles/bigdir/subdir2/subdir23/file3'
chgrp: invalid group: 'rdiff-backup_testfiles/bigdir/subdir3/subdir10/file31'
chown: invalid user: 'rdiff-backup_testfiles/corrupt_dest2/rdiff-backup-data/last-file-incremented.2002-07-23T17:08:00-07:00.data'
chgrp: invalid group: 'rdiff-backup_testfiles/corrupt_dest2/rdiff-backup-data/last-file-incremented.2002-07-23T17:08:00-07:00.data'
```